tests/Test/Xrefcheck/IgnoreRegexSpec.hs

```diff
@@ -0,0 +1,67 @@
+{- SPDX-FileCopyrightText: 2019 Serokell <https://serokell.io>
+ -
+ - SPDX-License-Identifier: MPL-2.0
+ -}
+
+module Test.Xrefcheck.IgnoreRegexSpec where
+
+import Data.List (isInfixOf)
+import Data.Yaml (decodeFileEither, prettyPrintParseException)
+import Test.Hspec (Spec, describe, it, shouldBe)
+
+import Xrefcheck.Config (Config(..))
+import Xrefcheck.Core
+import Xrefcheck.Progress (allowRewrite)
+import Xrefcheck.Scan (gatherRepoInfo, specificFormatsSupport)
+import Xrefcheck.Scanners.Markdown
+import Xrefcheck.Verify
+    (VerifyError, VerifyResult, WithReferenceLoc(..), verifyErrors, verifyRepo)
+
+spec :: Spec
+spec = do
+    describe "Regular expressions performance" $ do
+        let root = "tests/markdowns/without-annotations"
+        let showProgressBar = False
+        let formats = specificFormatsSupport [markdownSupport]
+        let verifyMode = ExternalOnlyMode
+
+        it "Check that only not matched links are verified" $ do
+            config <- readConfig "tests/configs/test-config.yaml"
+
+            repoInfo <- allowRewrite showProgressBar $ \rw ->
+                gatherRepoInfo rw formats (cTraversal config) root
+
+            verifyRes <- allowRewrite showProgressBar $ \rw ->
+                verifyRepo rw (cVerification config) verifyMode root repoInfo
+
+            let brokenLinks = pickBrokenLinks verifyRes
+
+            let matchedLinks =
+                    [ "https://bad.reference/"
+                    , "https://bad.reference"
+                    , "https://bad.external.reference/"
+                    , "https://bad.external.reference"
+                    , "https://bad.reference.ru"
+                    , "https://bad.reference.com"
+                    ]
+
+            let notMatchedLinks =
+                    [ "https://non-existent.reference/"
+                    , "https://bad.externall.reference"
+                    , "https://bad.reference.io"
+                    ]
+
+            (matchedLinks `isInfixOf` brokenLinks) `shouldBe` False
+            (notMatchedLinks `isInfixOf` brokenLinks) `shouldBe` True
```

Comment:
What do you want to test here?

Reply:
Due to the fact all links for these tests are invalid, they are either reported as invalid or ignored by the verifier.

```(matchedLinks `isInfixOf` brokenLinks) `shouldBe` False``` checks that links matched with regexs from `test-config.yaml` are not in broken links (not verified)

And ```(notMatchedLinks `isInfixOf` brokenLinks) `shouldBe` True``` checks that links that are not matched with regexs from `test-config.yaml` are contained in broken links (verified)

Reply:
As far as I understand:

* If `brokenLinks = [a, b, c]` and `matchedLinks = [a, d]`, then `isInfixOf` will return `False` and test will pass, despite presence of `a` should cause a failure. 
* If `brokenLinks = [a, b, c]` and `notMatchedLinks = [a, c]`, `isInfixOf` will return `False` because the latter is a subsequence of the former, but not a sublist, and the test with fail despite everything is good.

Actually, I would be a bit more verbose here and write something like

```haskell
forM links $ \link -> do
  when (link `elem` links0) $
    assertFailure $ "Link is considered broken despite it should be ignored: " <> show link
```

because having relatively proper error messages in tests is also valuable :slightly_smiling_face:.